games/TicTacToe/src/main.sw

```diff
@@ -0,0 +1,97 @@
+contract;
+
+dep interface;
+dep events;
+dep errors;
+dep utils;
+
+use interface::Game;
+use events::{GameDrawnEvent, GameWonEvent, NewGameEvent};
+use errors::Errors;
+use utils::{draw, win_check};
+use core::ops::Eq;
+use std::{auth::msg_sender, logging::log};
+
+impl<T> Eq for Option<T> {
+    fn eq(self, other: Self) -> bool {
+        match (self, other) {
+            (Option::None, Option::None) => true,
+            (Option::Some(T), Option::Some(T)) => true,
+            _ => false,
+        }
+    }
+}
+
+storage {
+    // Stores the players
+    player_one: Option<Identity> = Option::None,
+    player_two: Option<Identity> = Option::None,
+    // Stores each player's moves,
+    board: StorageMap<u64, Option<Identity>> = StorageMap {},
+    // Stores move counter
+    player_turn: Option<Identity> = Option::None,
```

Comment:
Should `player_turn` be an option? Is there a valid game state where the turn is undefined?

Reply:
As of now, yes because we use the `Option::None` state to stop the `move` function if there is either a winner or a draw, but this will be fixed in the next updates.

Reply:
As of now, yes because we use the `Option::None` state to stop the `move` function if there is either a winner or a draw, but this will be fixed in the next updates.